src/components/Filters/FilterModal.tsx

```diff
@@ -18,14 +18,22 @@ export function FilterModal<F>(props: FilterModalProps<F>) {
   const [modalFilter, setModalFilter] = useState<F>(filter);
 
   const updateFilter = useCallback((currentFilter: F, key: keyof F, value: any | undefined) => {
```

Comment:
Would like to try and make a reusable `updateFilter` function. There is enough logic in here now that we should share it between the `FilterModal` and `Filter`. 
They are accomplishing the same task, but just using different functions to set the filter: `setModalFilter` vs `onChange`.